lib/rom/sql/attribute.rb

```diff
@@ -235,6 +257,11 @@ def sql_expr
         @sql_expr ||= (meta[:sql_expr] || Sequel[to_sym])
       end
 
+      # @api private
+      def type_extensions
+        @type_extensions ||= TypeExtensions[unwrap_type(type)] || EMPTY_HASH
+      end
```

Comment:
Hmm, don't you think we could do it [here](https://github.com/rom-rb/rom/blob/master/lib/rom/schema.rb#L89-L91)? ie:

``` ruby
def self.attributes(attributes, attr_class)
  attributes.map { |type| attr_class.new(type, extensions: TypeExtensions[type]) }
end
```

This would require `TypeExtenions#[]` to unwrap when needed.